Carbon.Management/Functions/ConvertTo-CBase64.ps1

```diff
@@ -0,0 +1,197 @@
+
+function ConvertTo-CBase64
+{
+    <#
+    .SYNOPSIS
+    Base-64 encodes things.
+    
+    .DESCRIPTION
+    The `ConvertTo-CBase64` function base-64 encodes things. Pipe what you want to encode to `ConvertTo-CBase64`. The
+    function can encode:
+    
+    * [String]
+    * [byte]
+    * [char]
+    * Signed integers: [int16], [int], [int64] (i.e. [long])
+    * Unsigned integers: [uint16], [uint32], [uint64]
+    * Floating point numbers: [float], [double]
+    * [bool]
+
+    For each item piped to `ConvertTo-CBase64`, the function returns that item base-64 encoded.
+
+    If you pipe all bytes or all chars to `ConvertTo-CBase64`, it will encode all the bytes and chars together. This
+    allows you to do this:
+
+        [IO.File]::ReadAllBytes('some file') | ConverTo-CBase64
+
+    and get back a single string for all the bytes/chars.
+
+    By default, `ConvertTo-CBase64` uses UTF-8 encoding when converting strings to base-64 (this is the default encoding

```

Comment:
It does, but when I used that encoding by default, everything failed. I'll try again.

Reply:
That was merely a curiosity, I would prefer we defaulted to UTF-8.

Reply:
Since the strings internally to .NET are encoded as Unicode, if we encoded as UTF-8 there might be some loss. Also, the old `ConvertTo-CBase64` function used Unicode.

Reply:
That's a good point.